src/test/scala/chiselTests/DataPrint.scala

```diff
@@ -30,28 +30,37 @@ class DataPrintSpec extends ChiselFlatSpec with Matchers {
       FixedPoint(5.W, 3.BP).toString should be ("FixedPoint<5><<3>>")
       Vec(3, UInt(2.W)).toString should be ("UInt<2>[3]")
       EnumTest.Type().toString should be ("EnumTest")
-      (new BundleTest).toString should be ("BundleTest")
-    } }
+        (new BundleTest).toString should be ("BundleTest")
+      new Bundle { val a = UInt(8.W) }.toString should be ("AnonymousBundle")
+    }}
   }
 
   class BoundDataModule extends Module {  // not in the test to avoid anon naming suffixes
-    Wire(UInt()).toString should be("UInt(Wire in BoundDataModule)")
-    Reg(SInt()).toString should be("SInt(Reg in BoundDataModule)")
+    Wire(UInt()).toString should be("BoundDataModule: Wire[UInt]")
+    Reg(SInt()).toString should be("BoundDataModule: Reg[SInt]")
     val io = IO(Output(Bool()))  // needs a name so elaboration doesn't fail
-    io.toString should be("Bool(IO in unelaborated BoundDataModule)")
+    io.toString should be("BoundDataModule.io: IO[Bool]")
     val m = Mem(4, UInt(2.W))
-    m(2).toString should be("UInt<2>(MemPort in BoundDataModule)")
-    (2.U + 2.U).toString should be("UInt<2>(OpResult in BoundDataModule)")
-    Wire(Vec(3, UInt(2.W))).toString should be ("UInt<2>[3](Wire in BoundDataModule)")
+    m(2).toString should be("BoundDataModule: MemPort[UInt<2>]")
+    (2.U + 2.U).toString should be("BoundDataModule: OpResult[UInt<2>]")
+    Wire(Vec(3, UInt(2.W))).toString should be ("BoundDataModule: Wire[UInt<2>[3]]")
 
     class InnerModule extends Module {
       val io = IO(Output(new Bundle {
         val a = UInt(4.W)
       }))
     }
     val inner = Module(new InnerModule)
-    inner.clock.toString should be ("Clock(IO clock in InnerModule)")
-    inner.io.a.toString should be ("UInt<4>(IO io_a in InnerModule)")
+    inner.clock.toString should be ("InnerModule.clock.clock: IO[Clock]")
+    inner.io.a.toString should be ("InnerModule.io_a.a: IO[UInt<4>]")
+
+    class FooTypeTest extends Bundle {
+      val foo = Vec(2, UInt(8.W))
+      val fizz = UInt(8.W)
+    }
+    val tpe = new FooTypeTest
+    val fooio: FooTypeTest = IO(Input(tpe))
+    fooio.foo(0).toString should be ("BoundDataModule.foo[0]: IO[UInt<8>]")
```

Comment:
```suggestion
    fooio.foo(0).toString should be ("BoundDataModule.fooio.foo[0]: IO[UInt<8>]")
```

Shouldn't this print as the above? or `fooio_foo[0]`? There is no `foo` in BoundDataModule